### DISCUSSION TITLE
Allegato PDF a fattura

### DISCUSSION THREAD

**Discussion by teo256** [fattureincloud/api#120] [Q&A] [answered]
Buongiorno,

dove potrei trovare un esempio di codice in C# per l'upload di un allegato per una fattura?

Grazie.

EDIT:
Esempio trovato qui: [https://github.com/fattureincloud/fattureincloud-csharp-sdk/blob/master/docs/IssuedDocumentsApi.md#uploadissueddocumentattachment](url)

Ora però ricevo questo errore:

Eccezione generata: 'It.FattureInCloud.Sdk.Client.ApiException' in It.FattureInCloud.Sdk.dll
Exception when calling IssuedDocumentsApi.UploadIssuedDocumentAttachment: Error calling UploadIssuedDocumentAttachment: {"error":{"message":"Extension not valid.","validation_result":null}}
Status Code: 422

Questo è il codice:

```
            var filename = "2022_Fattura" + this.num_fatturaTextBox.Text + ".pdf";
            var attachment = new System.IO.MemoryStream(System.IO.File.ReadAllBytes("\\\\storage-04\\DB Lambda Studio\\PDF_Fatture\\2022\\2022_Fattura1.pdf"));
            try
            {
                // Upload Issued Document Attachment
                UploadIssuedDocumentAttachmentResponse result = apiInstance.UploadIssuedDocumentAttachment(companyId, filename, attachment);
                Debug.WriteLine(result);
            }
            catch (ApiException e)
            {
                Debug.Print("Exception when calling IssuedDocumentsApi.UploadIssuedDocumentAttachment: " + e.Message);
                Debug.Print("Status Code: " + e.ErrorCode);
                Debug.Print(e.StackTrace);
```

**Comment by emajo** [comment, marked as answer]
Ciao @teo256, ti chiederei di provare nel seguente modo:
`var attachment = File.Open(@"\\storage-04\DB Lambda Studio\PDF_Fatture\2022\2022_Fattura1.pdf", FileMode.Open);`
dovrebbe funzionare tutto come previsto.

Mi faccia sapere se funziona,
grazie,
Emanuele

**Reply by teo256** [reply]
Perfetto, funziona!!

Grazie mille (nuovamente) per l'aiuto!!
